src/Api/Controller/ListPostsController.php

```diff
@@ -93,6 +93,14 @@ protected function data(ServerRequestInterface $request, Document $document)
             $results->areMoreResults() ? null : 0
         );
 
+        if (! in_array('discussion', $include)) {
```

Comment:
This is concerning to me, doesn't this mean that we'd always be loading the discussion, even not when requested? 

Shouldn't we be checking that `'discussion.*'` is in the array? We might want to add a util for `AbstractSerializeController` that allow wildcard checks for inclusion in `$include`. Or maybe there's some PHP functio n already out there?

Reply:
that's the point, we need to always eager load it, because the PostSerializer which calls the PostPolicy uses the `$post->discussion` relationship, so you end up with either tens of database queries for each discussion, or one that loads them all.

The key difference to note here between including a relationship in a request and eager loading, is that whether you eager load the relationship or not, it will only be added to the JSON response if it was included in the request.

Reply:
Ah, got it. Can you add a comment explaining this? It's a bit counterintuitive from just looking at the code

Reply:
Sure